non-deterministic test failures zstd 1.4.9 on i686-linux

Hello!

Over at GNU Guix we are getting non-deterministic test failures for zstd 1.4.9 on i686-linux.

log:
```
roundTripTest: datagen -g8M  | zstd -v19 -T0 --long | zstd -d19 -T0 --long

*** zstd command line interface 32-bits v1.4.9, by Yann Collet ***
Note: 32 physical core(s) detected 
zstd: error 11 : Allocation error : not enough memory 
zstd: /*stdin*\: unexpected end of file 
Files tmp1 and tmp2 differ
make[1]: *** [Makefile:329: test-zstd] Error 1
```

Full log: https://ci.guix.gnu.org/build/378399/log/raw

To reproduce:
- Install GNU Guix (https://guix.gnu.org/manual/en/html_node/Binary-Installation.html#Binary-Installation)
- Run: `guix pull --commit=6f873731a030dd7ecbd8a5e756b38b26306f6966`
- Run: `guix build --system=i686-linux -K zstd@1.4.9` (few times, until the non-deterministic error is met)

Thanks

As another data point I did not run into this issue on Guix when building on my single-core powerpc iBook G4, but when the test is modified to use compression level 22 I actually went OOM. For the record I'm running with 1.5GB of RAM and 2.5GB of swap, 1GB of which is zram.

roundTripTest: datagen -g8M  | zstd -v22 -T0 --long | zstd -d22 -T0 --long

*** zstd command line interface 32-bits v1.4.9, by Yann Collet ***
Warning : compression level higher than max, reduced to 19
Note: 1 physical core(s) detected
zstd: error 11 : Allocation error : not enough memory
zstd: /*stdin*\: unexpected end of file
Files tmp1 and tmp2 differ
make[1]: *** [Makefile:329: test-zstd] Error 1

--- a/tests/playTests.sh
+++ b/tests/playTests.sh
@@ -1225,7 +1225,7 @@ then
     println "\n===>  zstdmt round-trip tests "
     roundTripTest -g4M "1 -T0"
     roundTripTest -g8M "3 -T2"
-    roundTripTest -g8M "19 -T0 --long"
+    roundTripTest -g8M "19 --long"
     roundTripTest -g8000K "2 --threads=2"
     fileRoundTripTest -g4M "19 -T2 -B1M"
 

